applications/RANSApplication/python_scripts/formulations/turbulence_models/scalar_turbulence_model_rans_formulation.py

```diff
@@ -0,0 +1,208 @@
+
+# import kratos
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+
+# import formulation interface
+from KratosMultiphysics.RANSApplication.formulations.rans_formulation import RansFormulation
+
+# import utilities
+from KratosMultiphysics.RANSApplication.formulations.utilities import CreateRansFormulationModelPart
+from KratosMultiphysics.RANSApplication.formulations.utilities import CreateBlockBuilderAndSolver
+from KratosMultiphysics.RANSApplication.formulations.utilities import InitializePeriodicConditions
+from KratosMultiphysics.RANSApplication.formulations.utilities import GetBoundaryFlags
+from KratosMultiphysics.RANSApplication.formulations.utilities import GetKratosObjectType
+from KratosMultiphysics.RANSApplication.formulations.utilities import CalculateNormalsOnConditions
+from KratosMultiphysics.RANSApplication.formulations.utilities import InitializeYPlusVariablesInConditions
+
+class ScalarTurbulenceModelRansFormulation(RansFormulation):
+    def __init__(self, model_part, settings):
+        """Scalar turbulence model formulation base class
+
+        This solves the variable given in self.GetSolvingVariable(), using element and conditions
+        having prefixes provided by self.GetElementNamePrefix() and self.GetConditionNamePrefix()
+
+        If wall functions are used, then self.GetConditionNamePrefix() should return non-empty prefix
+        otherwise it should be empty.
+
+        Args:
+            model_part (Kratos.ModelPart): ModelPart to be used in the formulation.
+            settings (Kratos.Parameters): Settings to be used in the formulation.
+        """
+        super().__init__(model_part, settings)
+
+        defaults = Kratos.Parameters(r"""{
+            "relative_tolerance"    : 1e-3,
+            "absolute_tolerance"    : 1e-5,
+            "max_iterations"        : 200,
+            "relaxation_factor"     : 0.5,
+            "echo_level"            : 2,
+            "linear_solver_settings": {
+                "solver_type"  : "amgcl"
+            },
+            "boundary_flags": ["INLET", "STRUCTURE"]
+        }""")
+
+        settings = self.GetParameters()
+        settings.ValidateAndAssignDefaults(defaults)
+        self.echo_level = settings["echo_level"].GetInt()
+
+    @abstractmethod
+    def GetSolvingVariable(self):
+        pass
+
+    @abstractmethod
+    def GetElementNamePrefix(self):
+        pass
+
+    @abstractmethod
+    def GetConditionNamePrefix(self):
+        pass
+
+    def PrepareModelPart(self):
+        self.turbulence_model_part = CreateRansFormulationModelPart(
+            self.GetComputingModelPart(),
+            self.__class__.__name__,
+            self.GetDomainSize(),
+            self.element_name,
+            self.condition_name)
+
+        Kratos.Logger.PrintInfo(self.__class__.__name__,
+                                "Created formulation model part.")
+
+    def Initialize(self):
```

Comment:
The purpose of this  `GetKratosObjectType` function is not clear to me. Can you comment something about it?

Reply:
It is used to get the environment specific ObjectType of request. Following is an example:
https://github.com/KratosMultiphysics/Kratos/blob/a35bf7bd8cbbaa6603182c1762ac59524520e1b9/applications/RANSApplication/python_scripts/formulations/utilities.py#L25-L28

Using above if following line is executed,
`linear_solver = GetKratosObjectType("LinearSolverFactory")(linear_solver_settings)`

will return OMP linear solver factory if run in OMP, or will return trilinos linear solver factory if run in MPI. 

I am using this, because `RansFormulation` does not need to know what the environment is, they are handled by `RansCoupledSolver` and utilities. This allows me to keep `RansFormulations` clean and simple. Otherwise I would have to create trilinos specific formulation for each case, and when `RansFormulation` is added to another `RansFormulation` I have to use if conditions there to add correct OMP or MPI version

Reply:
OK. Now I understand and makes perfect sense. However, I suggest to rename it to `GetKratosObject`. Note that you are not returning a type but an instance of the object.